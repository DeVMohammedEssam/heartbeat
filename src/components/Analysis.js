import React from 'react';
import Navbar from './layout/Navbar';
import axios from "axios"
import {url} from "../config"
class Analysis extends React.Component{
    state={
      postive:0
      ,negative:0,
      all:0
    }
       componentDidMount(){
        axios.get(url+"analysis").then((response)=>{
                this.setState({
                    ...response.data
                })
        })
    }
  render(){
    return (
        <React.Fragment>
            <Navbar />
            <section className="analysis">
                <h3 className="analysis__title">Analysis</h3>
                <div className="analysis__cards-container">
                    <div className="analysis__card">
                        <div className="analysis__card__icon-container">
                            <img src="/images/users-group.png" alt="" />
                        </div>
                        <div className="analysis__card__number">{this.state.all}</div>
                    </div>
                    <div className="analysis__card">
                        <div className="analysis__card__icon-container">
                            <img src="/images/patient.png" alt="" />
                        </div>
                        <div className="analysis__card__number">{this.state.postive}</div>
                    </div>
                    <div className="analysis__card">
                        <div className="analysis__card__icon-container">
                            <img src="/images/verified.png" alt="" />
                        </div>
                        <div className="analysis__card__number">{this.state.negative}</div>
                    </div>
                </div>
            </section>
        </React.Fragment>
    );
  }
}

export default Analysis;