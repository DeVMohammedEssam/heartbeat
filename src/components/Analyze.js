import React, { Component } from "react";
import Navbar from "./layout/Navbar";
import HeartLoader from "./layout/HeartLoader";
import {connect} from "react-redux"
import { startAnalyze } from "../redux/actions/index";
import validateAnalyzeInputs from "../validation/analyze";
export class Analyze extends Component {
  state = {
    analyzeClicked: false,
    data: {
      age: "",
      sex: "male",
      cp: "",
      trestbps: "",
      chol: "",
      fbs: "",
      restecg: "",
      thalach: "",
      exang: "",
      oldpeak: "",
      slope: "",
      ca: "",
      thal: ""
    },
    errors: {}
  };
  handleSubmit = e => {
    e.preventDefault();
    const { isValid, errors } = validateAnalyzeInputs(this.state.data);

    if (!isValid) {
      this.setState(() => ({ errors }));
    } else {
      this.setState(() => ({ analyzeClicked: true })); //show loader

    this.props.dispatch(startAnalyze(this.state.data)).then(()=>{  //Add Analyze Data to StartAnalyze Function as one param
            
                       this.setState(() => ({ analyzeClicked: false })); 
                    }) 
    }
  };
  handleChange = e => {
    const { name, value } = e.target;
    this.setState(() => ({ data: { ...this.state.data, [name]: value } }));
  };
  render() {
    return (
      <React.Fragment>
        <Navbar />
        <div className="analyze row">
          {this.state.analyzeClicked ? (
            /* data submitted and waiting for result */
            <div className="col-12 col-lg-8 analyze__loader">
              <HeartLoader />
            </div>
          ) : (
            /* did not submit yet */
            <form
              className="analyze__check-form col-12 col-lg-8 "
              onSubmit={this.handleSubmit}
            >
              <h4 className="analyze__check-title">check your heart</h4>
              <div className="analyze__inputs-container ">
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="age"
                  name="age"
                  value={this.state.data.age}
                />
                <select
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="sex"
                  name="sex"
                  defaultValue={this.state.data.sex}
                >
                  <option value="male">male</option>
                  <option value="female">female</option>
                </select>
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="cp"
                  name="cp"
                  value={this.state.data.cp}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="trestbps"
                  name="trestbps"
                  value={this.state.data.trestbps}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="chol"
                  name="chol"
                  value={this.state.data.chol}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="fbs"
                  name="fbs"
                  value={this.state.data.fbs}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="restecg"
                  name="restecg"
                  value={this.state.data.restecg}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="thalach"
                  name="thalach"
                  value={this.state.data.thalach}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="exang"
                  name="exang"
                  value={this.state.data.exang}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="oldpeak"
                  name="oldpeak"
                  value={this.state.data.oldpeak}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="slope"
                  name="slope"
                  value={this.state.data.slope}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="ca"
                  name="ca"
                  value={this.state.data.ca}
                />
                <input
                  type="text"
                  className="analyze__input custom-input "
                  onChange={this.handleChange}
                  placeholder="thal"
                  name="thal"
                  value={this.state.data.thal}
                />
              </div>

              <input
                type="submit"
                value="Analyize Now"
                className=" analyze__analye-btn custom-btn custom-btn--lightBlue"
              />
            </form>
          )}

          <div className=" col-12  col-lg-3">
            <aside className="analyze__note">
              <p className="analyze__note__text">
                {" "}
                adi cumque earum provident sint accusantium odit, eveniet animi
                perferendis.
              </p>
              <p className="analyze__note__text">
                sasperiores cumque earum provident sint accusantium odit,
                eveniet animi perferendis.
              </p>
              <p className="analyze__note__text">
                {" "}
                as asperiores cumque earum provident sint accusantium odit,
                eveniet animi perferendis.
              </p>
              <p className="analyze__note__text">
                {" "}
                cumque earum provident sint accusantium odit, eveniet animi
                perferendis.
              </p>
            </aside>
            <aside className="analyze__note-bottom">
              <p className="analyze__note__text">
                {" "}
                provident sint accusantium odit, eveniet animi perferendis.
              </p>
              <p className="analyze__note__text">
                {" "}
                provident sint accusantium odit, eveniet animi perferendis.
              </p>
            </aside>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default connect()(Analyze);
