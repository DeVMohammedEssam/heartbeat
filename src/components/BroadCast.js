import React, { Component } from 'react';
import Navbar from "./layout/Navbar";
import {sendNotification} from "../redux/actions/index"
class BroadCast extends Component {
    state = {
        sendNowClicked: false
    }
    handleSubmit = (e) => {
        e.preventDefault();
        this.setState(() => ({ sendNowClicked: true }));
 /*
        this.props.dispatch(sendNotification(data)).then(()=>{
 this.setState({sendNowClicked:false})
        })
*/
    }
    render() {
        return (
            <React.Fragment>
                <Navbar />
                <form className="broadcast" onSubmit={this.handleSubmit}>
                    <h3 className="broadcast__title">Broadcast</h3>
                    <div className="broadcast__content">
                        <div className="broadcast__textbox">
                            <textarea className="custom-input" name="broadcast-message" placeholder="Send message"></textarea>
                            {this.state.sendNowClicked && (
                                <span className="done-image-container"></span>
                            )}
                        </div>
                        <div className="broadcast__send-to ">
                            <div className="wrapper">

                                <select name="sentTo" className=" custom-btn btn--darkGrey">
                                    <option value="patients">patients</option>
                                    <option value="doctors">negative people</option>
                                </select>
                                <span className="subtitle">message will be sent to: 26,381 users</span>
                            </div>
                            <input className="broadcast__send-now custom-input " type="submit" value="Send Now" />
                        </div>
                    </div>
                </form>
            </React.Fragment>
        );
    }
}

export default BroadCast;