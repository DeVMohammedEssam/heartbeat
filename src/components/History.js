import React, { Component } from 'react';
import ResultCard from "./layout/ResultCard";
import { connect } from "react-redux"
import Navbar from './layout/Navbar';
import { startGetHistory } from "../redux/actions/index"
class History extends Component {
    state = { analysis: [] }
    componentDidMount() {
        this.props.dispatch(startGetHistory()).then((response) => {
            // console.log(response.data.anaylsis)
            this.setState({ analysis: response.data.anaylsis })
        })
    }
    render() {
        return (
            <React.Fragment>
                <Navbar />
                <div className="history">
                    <h3 className="history__title">History</h3>
                    <div className="row history__content mt-5">
                        <div className=" col-12 col-lg-3 order-2 order-lg-1 mt-5 mt-lg-0">
                            <div className="history__left-widget">

                                {this.state.analysis.map((analyze) => (

                                    <ResultCard status={analyze.result ? "success" : "fail"} date={() => { return analyze.date.getFullYear() + "/" + analyze.date.getMonth() + "/" + analyze.date.getDay() }} />

                                ))}

                            </div>
                        </div>
                        <div className="col-12 col-lg-9 order-1 order-lg-2">
                            <div className="history__readings ">
                                {this.state.analysis.map((analyze) => (
                                    <div className="reading">
                                        <div className="row">
                                            <div className="col-md-3">
                                                <p className="reading__text"> age</p> <span className="reading__percentage">{analyze.age}</span>
                                            </div>
                                            <div className="col-md-3">  <p className="reading__text"> sex</p>  <span className="reading__percentage">{analyze.sex}</span>   </div>
                                            <div className="col-md-3">   <p className="reading__text"> cp</p> <span className="reading__percentage">{analyze.cp}</span>   </div>
                                            <div className="col-md-3">  <p className="reading__text"> chol</p> <span className="reading__percentage">{analyze.chol}</span>   </div>
                                            <div className="col-md-3">   <p className="reading__text"> fbs</p> <span className="reading__percentage">{analyze.fbs}</span>   </div>
                                            <div className="col-md-3">   <p className="reading__text"> restecg</p> <span className="reading__percentage">{analyze.restecg}</span>   </div>
                                            <div className="col-md-3">   <p className="reading__text"> thalach</p> <span className="reading__percentage">{analyze.thalach}</span>   </div>
                                            <div className="col-md-3">   <p className="reading__text"> exang</p> <span className="reading__percentage">{analyze.exang}</span> </div>
                                            <div className="col-md-3">   <p className="reading__text"> oldpeak</p> <span className="reading__percentage">{analyze.oldpeak}</span>   </div>
                                            <div className="col-md-3">   <p className="reading__text"> slope</p> <span className="reading__percentage">{analyze.slope}</span>   </div>
                                            <div className="col-md-3">  <p className="reading__text"> cas</p> <span className="reading__percentage">{analyze.cas}</span>   </div>
                                            <div className="col-md-3">  <p className="reading__text"> thal</p> <span className="reading__percentage">{analyze.thal}</span>   </div>
                                            <div className="col-md-3"> <p className="reading__text"> class</p> <span className="reading__percentage">{analyze.class}</span>   </div>
                                        </div>


                                    </div>
                                ))}


                            </div>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        );
    }
}

export default connect()(History);