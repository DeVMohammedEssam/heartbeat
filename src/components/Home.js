import React, { Component } from 'react';
import Navbar from "./layout/Navbar";
class Home extends Component {
    state = {}
    render() {
        return (
            <React.Fragment>
                
                <Navbar />
                <div className="home">
                    <div className="home__cover"></div>
                    <p className="home__text">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium corporis doloribus voluptatem nemo, asperiores itaque culpa laudantium tempore obcaecati architecto.</p>
                    <div className="home__heart-logo-container">
                        <img src="images/heart-icon.png " className="home__heart-logo " alt="" />
                        <span className="home__heart-logo-subtitle ">Our mission</span>

                    </div>
                    <p className="home__text">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad voluptate accusantium ratione iure quaerat voluptates, harum porro magni earum illum distinctio, voluptatibus nostrum iusto. Impedit praesentium voluptatum amet ex explicabo commodi ea omnis maxime sed. Iusto ducimus dolor odio magnam doloribus architecto illo ea! Nulla aliquid ab accusamus non ipsa?</p>
                    <div className="home__right-widget-container">
                        <img className="home__right-widget-icon" src="/images/right-widget-1.png" alt="" />
                        <img className="home__right-widget-icon" src="/images/arrow.png" alt="" />
                        <img className="home__right-widget-icon" src="/images/right-widget-2.png" alt="" />
                        <img className="home__right-widget-icon" src="/images/arrow.png" alt="" />
                        <img className="home__right-widget-icon" src="/images/right-widget-3.png" alt="" />
                    </div>
                    <div className="home__checknow-logo-container">
                        <img className="home__checknow-logo" src="/images/contract.png" alt="" />
                        <span>Check Now</span>
                    </div>

                </div>
            </React.Fragment>
        );
    }
}

export default Home;