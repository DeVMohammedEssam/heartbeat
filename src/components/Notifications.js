import React, { Component } from 'react';
import Navbar from './layout/Navbar';
import NotificationCard from "./layout/NotificationCard";
import {startGetNotifications} from "../redux/actions/index"
import {connect} from "react-redux"
class Notifications extends Component {
    state = {notifications:[]}
    componentDidMount(){
        this.props.dispatch(startGetNotifications()).then((response)=>{
            this.setState({notifications:response.data.notifications})
        })
    }
    render() {
        return (
            <React.Fragment>
                <Navbar />
                <section className="notifications">
                    <h3 className="notifications__title">           {this.props.user.type === 1 ?"broadcast history":"Notifications"}</h3>
                    <div className="notifications__list">
                        {this.state.notifications.map((notification)=>(
                        <NotificationCard sender={notification.from.email} message={notification.message} />
                        ))}
                    
                    </div>
                </section>
            </React.Fragment>
        );
    }
}
const mapStateToProps=(state)=>({
    user:state.user
})

export default connect(mapStateToProps)(Notifications);