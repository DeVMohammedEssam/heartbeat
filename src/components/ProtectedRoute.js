import React from 'react';
import { Route } from "react-router-dom";
import Sorry from "./Sorry"
const ProtectedRoute = ({ component: Component, rest }) => {
    return (
        <Route {...rest} render={(props) => {
            if (localStorage.getItem("userToken")) {//is logged in
                return (<Component />)
            } else {//is not logged in
         return (<Sorry {...props} />)
               // props.history.push("/");
            }
        }} />
    );
}

export default ProtectedRoute;