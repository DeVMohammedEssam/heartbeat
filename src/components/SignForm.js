import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { login } from "../redux/actions/user";
import { url } from "../config";
import axios from "axios";
import validateLoginInputs from "../validation/login";
import validateRegisterInputs from "../validation/register";

export class SignForm extends Component {
  state = {
    data: {
      email: "",
      password: "",
      password2: "",
      name: "",
      type: this.props.sign === "in" ? undefined : "0"
    },
    success: "",
    errors: {}
  };
  handleInputChange = e => {
    const name = e.target.name,
      value = e.target.value;
    let data = this.state.data;
    data[name] = value;

    this.setState(() => ({ data: { ...data } }));
  };

  handleSubmit = e => {
    e.preventDefault();
    if (this.props.sign === "up") {
      //sign up validation
      const { isValid, errors } = validateRegisterInputs(this.state.data);
      if (!isValid) this.setState(() => ({ errors }));
      else {
        axios
          .post(`${url}user/signup`, {
            data: this.state.data
          })
          .then(data => {
            if (data.data.success) {
              this.setState(() => ({
                success: "you Signed up successfully please login :)"
              }));
              setTimeout(() => {
                this.props.history.push("/");
              }, 2000);
            } else {
              this.setState(() => ({
                errors: { email: "email is already used" }
              }));
            }
          });
      }
    } else {
      //if the page was login
      //login validation
      const { isValid, errors } = validateLoginInputs({
        email: this.state.data.email,
        password: this.state.data.password
      });
      if (!isValid) {
        this.setState(() => ({ errors }));
      } else {
        //sending login request
        axios
          .post(`${url}user/login`, {
            data: this.state.data
          })
          .then(data => {
            if (data.data.success) {
              localStorage.setItem("userToken", data.data.token);
              this.props.dispatch(login()).then(() => {
                this.props.history.push("/analyze");
              });
            } else {
              this.setState(() => ({
                errors: { wrongEmailOrPassword: "wrong email or password" }
              }));
            }
          });
      }
    }
  };

  render() {
    return (
      <div>
        <form noValidate className="sign-form" onSubmit={this.handleSubmit}>
          {this.state.success && (
            <div className="alert alert-success flash-message">
              {this.state.success}
            </div>
          )}
          {this.state.error && (
            <div className="alert alert-danger flash-message">
              {this.state.error}
            </div>
          )}
          <div className="sign-form__inputs-container ">
            <input
              className="custom-input"
              type="email"
              name="email"
              placeholder="Email"
              value={this.state.data.email}
              onChange={this.handleInputChange}
              required
            />
            <input
              className="custom-input"
              type="password"
              name="password"
              placeholder="password"
              value={this.state.data.password}
              onChange={this.handleInputChange}
              required
            />

            {this.props.sign === "up" && (
              <React.Fragment>
                <input
                  className="custom-input"
                  type="name"
                  name="name"
                  placeholder="Name"
                  value={this.state.data.name}
                  onChange={this.handleInputChange}
                  required
                />
                <input
                  className="custom-input"
                  type="password"
                  name="password2"
                  placeholder="confirm password"
                  value={this.state.data.password2}
                  onChange={this.handleInputChange}
                  required
                />
                <select
                  className="custom-input"
                  name="type"
                  defaultValue={this.state.data.type}
                  onChange={this.handleInputChange}
                >
                  <option value="0">Patient</option>
                  <option value="1">Doctor</option>
                </select>
              </React.Fragment>
            )}
            <div className="btns-wrapper">
              <Link to="/" className="return-home">
                <i />
                <span>retrun to home</span>
              </Link>
              <input
                type="submit"
                className="custom-btn custom-btn--lightBlue"
                value={this.props.sign === "in" ? "sign in" : "sign up"}
              />
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default connect()(withRouter(SignForm));
