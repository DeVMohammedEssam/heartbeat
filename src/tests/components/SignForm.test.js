import React from "react";
import { SignForm } from "../../components/SignForm";
import { shallow } from "enzyme";
//login form
test("should render sign in form when making sign prop equal in", () => {
  const wrapper = shallow(<SignForm sign="in" />);
  expect(wrapper).toMatchSnapshot();
});
test("should raise email is required error", () => {
  const wrapper = shallow(<SignForm sign="in" />);
  wrapper.setState({ data: { email: "" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.email).toBe("email is required");
});
test("should raise email is invalid error", () => {
  const wrapper = shallow(<SignForm sign="in" />);
  wrapper.setState({ data: { email: "invalidEmail" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.email).toBe("email is invalid");
});
test("should rasie password is required error", () => {
  const wrapper = shallow(<SignForm sign="in" />);
  wrapper.setState({ data: { password: "", email: "meemasfw@gmail.com" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe("password is required");
});
test("should raise no errors", () => {
  const wrapper = shallow(<SignForm sign="in" />);
  wrapper.setState({
    data: { password: "123456", email: "meemasfw@gmail.com" }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors).toEqual({});
});

//register form
test("should render sign up form when making sign prop equal up", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  expect(wrapper).toMatchSnapshot();
});

//required fields
test("should raise email is required error", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { email: "" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.email).toBe("email is required");
});

test("should rasie password is required error", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { password: "", email: "meemasfw@gmail.com" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe("password is required");
});
test("should rasie password confirmation is required error", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { password2: "" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password2).toBe("password confirmation is required");
});

//logics
test("should raise email is invalid error", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { email: "invalidEmail" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.email).toBe("email is invalid");
});

test("should raise password doesn't match confirmation error", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { password: "123456", password2: "123457" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password2).toBe("password does not match confirmation");
});

//chritereas

/*   if (!validator.isLength(data.name, { min: 2, max: 30 }))
  errors.name = "name must be between 2 and 30 characters";
  */

test("should raise name must be between 2 and 30 characters error when name letters less than 2 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { name: "a" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.name).toBe("name must be between 2 and 30 characters");
});
test("should raise name must be between 2 and 30 characters error when name letters more than 30 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({
    data: {
      name: "mohammed essam el deen mohammed" //31
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.name).toBe("name must be between 2 and 30 characters");
});
test("should raise no errors when letters number is 2 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({
    data: {
      name: "mo"
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.name).toBe(undefined);
});
test("should raise no errors when letters number is 30 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({
    data: {
      name: "mohammed essam el deen mohamme" // 30
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.name).toBe(undefined);
});
test("should raise no errors when name letters number between 2 and 30 letters", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({
    data: {
      name: "mohammed essam" //14
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.name).toBe(undefined);
});
/*
  
if (!validator.isLength(data.password, { min: 6, max: 50 }))
  errors.password = "password must be between 6 and 50 characters"; */
test("should raise password must be between 6 and 50 characters error when password letters less than 6 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({ data: { password: "12345" } });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe("password must be between 6 and 50 characters");
});
test("should raise password must be between 6 and 50 characters error when password letters more than 50 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  let pass = "";
  for (let i = 0; i < 51; i++) {
    pass += "w";
  }
  wrapper.setState({
    data: {
      password: pass
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe("password must be between 6 and 50 characters");
});
test("should raise no errors when password letters number is 6 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({
    data: {
      password: "123456"
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe(undefined);
});
test("should raise no errors when password letters number is 50 ", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  let pass = "";
  for (let i = 0; i < 50; i++) {
    pass += "w";
  }
  wrapper.setState({
    data: {
      password: pass // 50
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe(undefined);
});
test("should raise no errors when password letters number between 6 and 50 letters", () => {
  const wrapper = shallow(<SignForm sign="up" />);
  wrapper.setState({
    data: {
      password: "mohammed essam" //14
    }
  });
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });
  const errors = wrapper.state("errors");
  expect(errors.password).toBe(undefined);
});
