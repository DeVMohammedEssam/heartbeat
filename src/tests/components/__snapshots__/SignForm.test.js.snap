// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render sign in form when making sign prop equal in 1`] = `
<div>
  <form
    className="sign-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="sign-form__inputs-container "
    >
      <input
        className="custom-input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        className="custom-input"
        name="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
      <input
        className="custom-input"
        name="password2"
        onChange={[Function]}
        placeholder="confirm password"
        required={true}
        type="password"
        value=""
      />
      <div
        className="btns-wrapper"
      >
        <Link
          className="return-home"
          to="/"
        >
          <i />
          <span>
            retrun to home
          </span>
        </Link>
        <input
          className="custom-btn custom-btn--lightBlue"
          type="submit"
          value="sign in"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`should render sign up form when making sign prop equal up 1`] = `
<div>
  <form
    className="sign-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="sign-form__inputs-container "
    >
      <input
        className="custom-input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        className="custom-input"
        name="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
      <input
        className="custom-input"
        name="password2"
        onChange={[Function]}
        placeholder="confirm password"
        required={true}
        type="password"
        value=""
      />
      <input
        className="custom-input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="name"
        value=""
      />
      <select
        className="custom-input"
        defaultValue="0"
        name="type"
        onChange={[Function]}
      >
        <option
          value="0"
        >
          Patient
        </option>
        <option
          value="1"
        >
          Doctor
        </option>
      </select>
      <div
        className="btns-wrapper"
      >
        <Link
          className="return-home"
          to="/"
        >
          <i />
          <span>
            retrun to home
          </span>
        </Link>
        <input
          className="custom-btn custom-btn--lightBlue"
          type="submit"
          value="sign up"
        />
      </div>
    </div>
  </form>
</div>
`;
